packages/convai-widget-core/src/components/Rating.tsx

```diff
@@ -0,0 +1,179 @@
+import { useSignal } from "@preact/signals";
+import { clsx } from "clsx";
+import { Icon, type IconName } from "./Icon";
+import { Signalish } from "../utils/signalish";
+
+const generateRatingValues = (min: number, max: number): number[] =>
+  Array.from({ length: max - min + 1 }, (_, i) => i + min);
+
+interface RatingIconProps {
+  isFilled: boolean;
+  iconName: IconName;
+}
+
+const RatingIcon = ({ isFilled, iconName }: RatingIconProps) => {
+  return (
+    <span
+      className={clsx(
+        "w-8 h-8 grid place-content-center text-base-primary",
+        !isFilled && "opacity-15"
+      )}
+    >
+      <Icon name={iconName} size="lg" />
+    </span>
+  );
+};
+
+interface RatingButtonProps {
+  value: number;
+  isFilled: boolean;
+  isHovered: boolean;
+  onClick: (value: number) => void;
+  onHover: (value: number) => void;
+  onKeyDown: (e: KeyboardEvent, value: number) => void;
+  iconName: IconName;
+}
+
+const RatingButton = ({
+  value,
+  isFilled,
+  isHovered,
+  onClick,
+  onHover,
+  onKeyDown,
+  iconName,
+}: RatingButtonProps) => {
+  const handleClick = (e: MouseEvent) => {
+    e.stopPropagation();
+    onClick(value);
+  };
+
+  const handleKeyDown = (e: KeyboardEvent) => {
+    onKeyDown(e, value);
+  };
+
+  return (
+    <span
+      role="radio"
+      aria-checked={isFilled}
+      aria-label={`${value} star${value !== 1 ? "s" : ""}`}
+      tabIndex={0}
+      onMouseEnter={() => onHover(value)}
+      onClick={handleClick}
+      onKeyDown={handleKeyDown}
+      className={clsx(
+        "w-8 h-8 grid place-content-center text-base-primary transition-opacity",
+        !isFilled && !isHovered && "opacity-15"
```

Comment:
Two things, let's add `rounded-full and focus-ring` to improve replace the default blue focus outline with the same style as other components use.

Also I'd suggest using `text-base-border` instead of `opacity-15` to not affect the visibility of the focus style:

```suggestion
        "w-8 h-8 grid place-content-center rounded-full focus-ring text-base-primary transition-opacity",
        !isFilled && !isHovered && "text-base-border"
```